### PR TITLE
LTP: Fix testcase sighold02

### PR DESCRIPTION
**Pull request by shaikshavali1**
This test case creates a child process using "FORK_OR_VFORK".
Fork behavior is not supported in sgx-lkl.
Hence, the below modifications are performed.

        1. Removed the code related to child process creation.
        2. combined parent and child functionality in a single process.

### CODE REVIEW

File: testcases/kernel/syscalls/sighold/sighold02.c [hunk 1]
```diff
@@ -61,6 +61,7 @@
 # define NUMSIGS NSIG
 #endif
 
+#define CODECOMMENT 0
```
Comment (davidchisnall) [inline testcases/kernel/syscalls/sighold/sighold02.c]: What is this?

File: testcases/kernel/syscalls/sighold/sighold02.c [hunk 2]
```diff
@@ -102,6 +103,7 @@ int main(int ac, char **av)
 	setup();
 
 	for (lc = 0; TEST_LOOPING(lc); lc++) {
+#if CODECOMMENT
```
Comment (davidchisnall) [inline testcases/kernel/syscalls/sighold/sighold02.c]: Please just use `#if 0`.  Everyone (including most editors) knows that this means disabled code.  Having to go and look for the definition of `CODECOMMENT` makes it harder to read.

File: testcases/kernel/syscalls/sighold/sighold02.c (hunk 1 above)
Comment (shaikshavali1) [reply testcases/kernel/syscalls/sighold/sighold02.c]: Removed.

File: testcases/kernel/syscalls/sighold/sighold02.c (hunk 2 above)
Comment (shaikshavali1) [reply testcases/kernel/syscalls/sighold/sighold02.c]: Code modifications are done.

**Review by davidchisnall — APPROVED** [review]

**Review by hukoyu — APPROVED** [review]
LGTM

**Pull request merged**